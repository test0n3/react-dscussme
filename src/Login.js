/** @jsx jsx */
import React from "react";
import { jsx } from "@emotion/core";

function Login({ label, placeholder, saveUser }) {
  const [currentUser, setCurrentUser] = React.useState("");

  function setUser(event) {
    event.preventDefault();
    saveUser(currentUser);
    setCurrentUser("");
  }

  function onChangeInputUser(event) {
    setCurrentUser(event.currentTarget.user);
  }
  return (
    <div>
      <div>
        <form onsubmit={setUser}>
          <label htmlFor="user">{label}</label>
          <input
            id="user"
            required="required"
            autocomplete="off"
            user="user"
            placeholder={placeholder}
            onChange={onChangeInputUser}
            value={currentUser}
          />
          {/* <input
           id="email"
           required="required"
           autocomplete="off"
           emailUser="email"
           placeholder={placeholder}
           
           
          
          /> */}
          <button>OK</button>
        </form>
      </div>
    </div>
  );
}

export default Login;
