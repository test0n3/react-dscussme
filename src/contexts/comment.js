import React from "react";

const CommentContext = React.createContext();

function CommentProvider({ comments, setComments, children }) {
  return (
    <CommentContext.Provider value={{ comments, setComments }}>
      {children}
    </CommentContext.Provider>
  );
}

function useComment() {
  return React.useContext(CommentContext);
}

export { CommentProvider, useComment };
