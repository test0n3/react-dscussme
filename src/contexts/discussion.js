import React from "react";

const DiscussionContext = React.createContext();

function DiscussionProvider({ discussions, setDiscussions, children }) {
  return (
    <DiscussionContext.Provider value={{ discussions, setDiscussions }}>
      {children}
    </DiscussionContext.Provider>
  );
}

function useDiscussion() {
  return React.useContext(DiscussionContext);
}

export { DiscussionProvider, useDiscussion };
