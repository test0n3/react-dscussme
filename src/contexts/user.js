import React from "react";

const UserContext = React.createContext();

function UserProvider({ user, setUser, children }) {
  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
}

function useUser() {
  return React.useContext(UserContext);
}

export { UserProvider, useUser };
