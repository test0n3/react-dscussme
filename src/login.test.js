import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Login from "./Login";
import { UserProvider } from "./contexts/user";

test("Login Form on Submit", () => {
  const placeholder = "write login";
  const { asFragment } = render(
    <UserProvider>
      <Login placeholder={placeholder} />
    </UserProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});