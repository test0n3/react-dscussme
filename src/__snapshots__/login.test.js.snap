// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form on Submit 1`] = `
<DocumentFragment>
  <div
    class="css-5xc5re-Login"
  >
    <div
      class="css-1azgr0o-Login"
    >
      <form
        class="css-c8sm1h-Login"
      >
        <label
          for="user"
        >
          Username
        </label>
        <input
          autocomplete="off"
          class="css-73r5xs-Login"
          id="user"
          placeholder="write login"
          required=""
          user="user"
          value=""
        />
        <br />
        <label
          for="email"
        >
          Email
        </label>
        <input
          autocomplete="off"
          class="css-73r5xs-Login"
          id="email"
          placeholder="write login"
          required=""
          type="email"
          value=""
        />
        <br />
        <button
          class="css-b7djqu-Login"
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
